test/fixtures/mandate-fixtures.js

```diff
@@ -0,0 +1,120 @@
+'use strict'
+
+const Payer = require('../../common/classes/Payer.class')
+const Mandate = require('../../common/classes/Mandate.class')
+const GatewayAccount = require('../../common/classes/GatewayAccount.class')
+const Service = require('../../common/classes/Service.class')
+
+const randomExternalId = () => Math.random().toString(36).substring(7)
+const randomNumber = () => Math.round(Math.random() * 10000) + 1
+const randomUrl = () => 'https://' + randomExternalId() + '.com'
+
+module.exports = {
+  validTokenExchangeResponse: (opts = {}) => {
+    const data = {
+      external_id: opts.external_id || randomExternalId(),
+      mandate_reference: opts.mandate_reference || 'buy Silvia a coffee',
+      type: opts.type || 'CHARGE',
+      state: opts.state || randomNumber(),
+      return_url: opts.return_url || randomUrl(),
+      gateway_account_external_id: opts.gateway_account_external_id || randomExternalId(),
+      gateway_account_id: opts.gateway_account_id || randomNumber()
+    }
+    return {
+      getPlain: () => {
+        return data
+      }
+    }
+  },
+  validGatewayAccountResponse: (opts = {}) => {
+    const data = {
+      gatewayAccountId: opts.gatewayAccountId || randomNumber(),
+      gatewayAccountExternalId: opts.gatewayAccountExternalId || randomExternalId(),
+      paymentMethod: opts.paymentMethod || 'DIRECT_DEBIT',
+      paymentProvider: opts.paymentProvider || 'SANDBOX',
+      type: opts.type || 'TEST'
+    }
+    return {
+      getPlain: () => {
+        return data
+      }
+    }
+  },
+  validMandateResponse: (opts = {}) => {
+    const data = {
+      external_id: opts.external_id || randomExternalId(),
+      return_url: opts.return_url || randomUrl(),
+      gateway_account_id: 23 || opts.gateway_account_id,
+      gateway_account_external_id: opts.gateway_account_external_id || randomExternalId(),
+      mandate_reference: opts.mandate_reference || 'buy Silvia a beer',
+      state: opts.state || 'started',
+      internal_state: opts.internal_state || 'AWAITING_DIRECT_DEBIT_DETAILS',
+      payer: opts.payer || null
+    }
+    return {
+      getPlain: () => {
+        return data
+      },
+      getObject: () => {
+        return new Mandate(data)
+      }
+    }
+  },
+  validCreatePayerResponse: (opts = {}) => {
+    const data = {
+      payer_external_id: opts.payer_external_id || randomExternalId()
+    }
+    return {
+      getPlain: () => {
+        return data
+      }
+    }
+  },
+
+  validPayer: (opts = {}) => {
```

Comment:
is `validPayer` representing a response from a dd-connector api call?

Reply:
yes looks like it, I haven't changed this as part of the PR though, I'll push a change to delete the file that was supposed to be renamed

Reply:
I don't think so, looks like it's just used in the setup mandate flow. Possibly doesn't really belong in this file, but that's a different change